crates/editor/src/editor.rs

```diff
@@ -3907,6 +3920,61 @@ impl Editor {
         }))
     }
 
+    pub fn show_signature_help(&mut self, _: &ShowSignatureHelp, cx: &mut ViewContext<Self>) {
+        if self.pending_rename.is_some() {
+            return;
+        }
+
+        let position = self.selections.newest_anchor().head();
+        let Some((buffer, buffer_position)) =
+            self.buffer.read(cx).text_anchor_for_position(position, cx)
+        else {
+            return;
+        };
+
+        let task = cx.spawn(move |this, mut cx| async move {
+            let markdown_task_result = this.update(&mut cx, |editor, cx| {
+                let project = editor.project.clone()?;
+                let (maybe_markdown, language_registry) = project.update(cx, |project, mut cx| {
+                    let language_registry = project.languages().clone();
+                    let maybe_markdown = project
+                        .signature_help(&buffer, buffer_position, &mut cx)
+                        .map(|signature_help| {
+                            create_signature_help_markdown_string(signature_help?)
+                        });
+                    (maybe_markdown, language_registry)
+                });
+                Some((maybe_markdown, language_registry))
+            });
+            let maybe_signature_help_popover =
+                if let Ok(Some((markdown, language_registry))) = markdown_task_result {
+                    let markdown = markdown.await;
+                    if let Some((markdown, markdown_highlights)) = markdown {
+                        let mut parsed_markdown =
+                            parse_markdown(markdown.as_str(), &language_registry, None).await;
+                        parsed_markdown.highlights = markdown_highlights;
+                        Some(SignatureHelpPopover {
+                            parsed_content: parsed_markdown,
+                        })
+                    } else {
+                        None
+                    }
+                } else {
+                    None
+                };
+            if let Some(signature_help_popover) = maybe_signature_help_popover {
```

Comment:
1. Let's not use `let _` and `.ok()` it instead in the production code, for better visibility. We could do `log_err` too, but usually it makes not much sense for "absent window" errors.
2. `this` could be named a bit more readable, that's `editor` presumably
3. Do something like
```rust
editor.update(cx, |editor, _| {
    editor.signature_help_state = maybe_signature_help_popover;
});
```
and rename `maybe_signature_help_popover` to just `signature_help_popover`?

Reply:
I fixed it on 06f1ad4f5